backend/pkg/kafka/config_helper.go

```diff
@@ -146,12 +147,34 @@ func NewKgoConfig(cfg *config.Kafka, logger *zap.Logger, hooks kgo.Hook) ([]kgo.
 
 		// AWS MSK IAM
 		if cfg.SASL.Mechanism == config.SASLMechanismAWSManagedStreamingIAM {
-			mechanism := aws.Auth{
-				AccessKey:    cfg.SASL.AWSMskIam.AccessKey,
-				SecretKey:    cfg.SASL.AWSMskIam.SecretKey,
-				SessionToken: cfg.SASL.AWSMskIam.SessionToken,
-				UserAgent:    cfg.SASL.AWSMskIam.UserAgent,
-			}.AsManagedStreamingIAMMechanism()
+			var mechanism sasl.Mechanism
+			if cfg.SASL.AWSMskIam.AccessKey == "" &&
+				cfg.SASL.AWSMskIam.SecretKey == "" &&
+				cfg.SASL.AWSMskIam.SessionToken == "" {
+				// support to use AWS IAM role
+				sess, err := session.NewSession()
+				if err != nil {
+					return nil, fmt.Errorf("unable to initialize aws session: %v", err)
+				}
+				val, err := sess.Config.Credentials.Get()
+				if err != nil {
+					return nil, fmt.Errorf("unable to get aws credentials: %v", err)
+				}
+				mechanism = aws.Auth{
+					AccessKey:    val.AccessKeyID,
+					SecretKey:    val.SecretAccessKey,
+					SessionToken: val.SessionToken,
+					UserAgent:    val.ProviderName,
+				}.AsManagedStreamingIAMMechanism()
+			} else {
```

Comment:
I'm not familiar how the AWS IAM works, could you add some comments here to describe why we we create a new "session" here in case no accesskey/secretkey/session token is provided? It looks like we would use ephemeral credentials pulled upon startup in that case or what is happening there exactly?

Reply:
Hello, 

I was able to finally build this locally and test on my EKS with IRSA setup, and it works!

```
❯ kubectl -n dev logs -f redpanda-console-5ffb68649-g8hhb
{"level":"info","ts":"2024-02-15T23:21:06.189Z","msg":"started Redpanda Console","version":"development","built_at":"<not set>"}
{"level":"info","ts":"2024-02-15T23:21:06.189Z","msg":"creating Kafka connect HTTP clients and testing connectivity to all clusters"}
{"level":"info","ts":"2024-02-15T23:21:06.190Z","msg":"tested Kafka connect cluster connectivity","successful_clusters":0,"failed_clusters":0}
{"level":"info","ts":"2024-02-15T23:21:06.190Z","msg":"successfully create Kafka connect service"}
{"level":"info","ts":"2024-02-15T23:21:06.318Z","msg":"connecting to Kafka seed brokers, trying to fetch cluster metadata"}
{"level":"info","ts":"2024-02-15T23:21:07.590Z","msg":"successfully connected to kafka cluster","advertised_broker_count":2,"topic_count":7,"controller_id":2,"kafka_version":"v3.6"}
{"level":"fatal","ts":"2024-02-15T23:21:07.596Z","msg":"failed to load index.html from embedded filesystem","error":"open index.html: file does not exist"}
```

With a config:

```
console:
  config:
    redpanda: {}
    kafka:
      brokers:
        - b-1.x1.kafka.us-east-1.amazonaws.com:9098
        - b-2.x1.us-east-1.amazonaws.com:9098
      clientId: redpanda-console
      schemaRegistry:
        enabled: false
      sasl:
        enabled: true
        mechanism: AWS_MSK_IAM
        awsMskIam: {}
      tls:
        enabled: true
        insecureSkipTlsVerify: true

# Additional environment variables for the Console Deployment
extraEnv:
  - name: AWS_REGION
    value: us-east-1
```

As you can see, under config `awsMskIam` I have not specified the accesskey/secretkey/session keys and let the IRSA do the sasl auth.

Reply:
This change actually was suggested here: https://github.com/twmb/franz-go/issues/39#issuecomment-1234999666

Reply:
If there are instructions on how to build and include UI, I'm happy to do that in the local docker file testing. Would love to run this for sometime ensuring the session credentials are renewed or if that needs handling as well.